KatelynKim/writegether
# feat: Story creation desktop view

## Description

Added the Story Creation page, desktop version.

This PR includes:
* Adding shadcn components (found under `/components/ui` (These do not need to be reviewed)
* Creating `StoryCreationForm` that uses a server action to post a story.
* Adding a callback function to `NextAuth` that attaches the user id from the jwt token to the user session so that it's accessible to server components. This required extending NextAuth's `Session` type as per this [doc](https://next-auth.js.org/getting-started/typescript#extend-default-interface-properties.

It does not include:
* Photo upload feature

Please focus the review on the story creation page and form, not the boilerplate code/configurations added by shadcn.

Closes #32

## Type of change

Please delete options that are not relevant and add additional details as necessary.

- [ ] Documentation update (changes to documentation or comments)

## How Has This Been Tested?

* On the top navigation bar, click on "Create" to go to the `/create` page.
* Type in the story title and opening segment.
* Click the post button
* Go to the [story collections](https://cloud.mongodb.com/v2/66b021d0c4943e680b3cabe7#/metrics/replicaSet/66b022d54159e857d2def42f/explorer/writegether-db/stories/find) in MongoDB and ensure that a new story document has been added with the correct fields.
